### ISSUE TITLE
Additional properties are incorrectly removed (removeAdditional)

### ISSUE THREAD

**Issue by webstacker** [cdimascio/express-openapi-validator#32]
I'm in the process of upgrading to 1.4.0 from 0.53.3 and many of our tests are failing. The following schema/body combination used to fail validation, which is the correct behaviour:

Request body schema:
```json
{
    "type": "object",
    "additionalProperties": false,
    "required": ["data"],
    "properties": {
        "data": {
            "type": "string"
        }
    }
}
```
Request body (this should fail):
```json
{
  "data": "foo",
  "not-allowed-property": "bar"
}
```

The addition of the "[removeAdditional](https://github.com/cdimascio/express-openapi-validator/blob/b69bcc5f850704f214f895f9eeb696a4bfb564c3/src/index.ts#L63)" property is causing the request body to be mutated prior to validation so the above example now appears to be valid, however this is what is actually being validated:

```json
{
  "data": "foo"
}
```

It would be nice to allow this to be configured, or even better, allow a raw options object to be passed through to the internal AJV instance. Is this something you would consider?

**Comment by sheldhur** [comment]
@webstacker you can use `additionalProperties: true`

[ajv#filtering-data](https://github.com/epoberezkin/ajv#filtering-data)

**Comment by webstacker** [comment]
@sheldhur Thanks for the suggestions, but it's not really an issue with the schema. Our schema explicitly says that it should not allow additional properties. The expected behaviour of that schema is that if additional properties are supplied, then it should fail. We can't test this expected behaviour.

Would there be any issue with allowing this to be configured in the options?

**Comment by cdimascio** [comment]
Thanks for the issue @webstacker. I will have a look at this over the weekend. This certainly should be an option. We'll get this in. Im currently contemplating whether to expose a new validator option or enable a passthrough to ajv opts. I'm currently leaning towards the former. The latter requires more cognitive load on the dev i.e. they must understand ajv and it's opts. also theynd may set opts that negatively affect the validator. I'm open to suggestions. Please fell free to share your thoughts.

**Comment by sheldhur** [comment]
@cdimascio I think about first variant, too, because `OpenAPI` now more limited than `JSONShema`.
Also required change `readOnly` keyword logic.

related #31

**Comment by cdimascio** [comment]
@sheldhur it feels the `removeAdditional` should be set to false on AJV

The additionalProperties use cases I've tested appear to work

When `removeAdditional` is false, these cases work correctly

e.g.  additionalProperties: false
```yaml
    NewPet:
      additionalProperties: false
       ...
```
```
curl --request POST \
  --url http://localhost:3000/v1/pets \
  --header 'content-type: application/json' \
  --data '{
        "name": "test", 
        "additional_prop_test": "test" 
}'

## returns 400
```
e.g.  additionalProperties: true
```yaml
    NewPet:
      additionalProperties: true
       ...
```
```
curl --request POST \
  --url http://localhost:3000/v1/pets \
  --header 'content-type: application/json' \
  --data '{
        "name": "test", 
        "additional_prop_test": "test" 
}'

## returns 200
```

e.g. additionalProperties
```yaml
    NewPet:
      additionalProperties:
        type: string 
      ...
```

```
curl --request POST \
  --url http://localhost:3000/v1/pets \
  --header 'content-type: application/json' \
  --data '{
        "name": "test", 
        "additional_prop_test": "test" 
}'

## returns 200
```


Let me know if I'm missing something.

**Comment by cdimascio** [comment]
fixed in v1.5.0
